src/message.c

```diff
@@ -536,33 +677,53 @@ int message_set_properties(MESSAGE_HANDLE message, PROPERTIES_HANDLE properties)
 {
 	int result;
 
-	if ((message == NULL) ||
-		(properties == NULL))
+	if (message == NULL)
 	{
-		LogError("Bad arguments: message = %p, properties = %p",
-			message, properties);
+		/* Codes_SRS_MESSAGE_01_054: [ If `message` is NULL, `message_set_properties` shall fail and return a non-zero value. ]*/
+		LogError("NULL message");
 		result = __FAILURE__;
 	}
 	else
 	{
-		PROPERTIES_HANDLE new_properties;
-
-		new_properties = properties_clone(properties);
-		if (new_properties == NULL)
-		{
-			LogError("Cannot clone message properties");
-			result = __FAILURE__;
-		}
-		else
+		if (properties == NULL)
 		{
+			/* Codes_SRS_MESSAGE_01_147: [ If `properties` is NULL, the previously stored message properties associated with `message` shall be freed. ]*/
 			if (message->properties != NULL)
 			{
 				properties_destroy(message->properties);
```

Comment:
Fixed.

---
In reply to: [120403958](https://github.com/Azure/azure-uamqp-c/pull/128#discussion_r120403958) [](ancestors = 120403958)